File: algorithms/O/Orderly Queue/Orderly Queue.java
// Time O(n)
// Space O(n)
class Solution {
    public String orderlyQueue(String s, int k) {
        int n = s.length();
        String ans = "";
        if (k == 1){
            s+=s; // add itself again
            for (int i = 0; i < n; i++) if (ans.isEmpty() || s.substring(i, i+n).compareTo(ans) < 0){
                ans = s.substring(i, i+n);
            }
        }else{
            char[] arr = s.toCharArray();
            Arrays.sort(arr);
            ans = String.valueOf(arr);
        }
        return ans;
    }
}

File: algorithms/O/Orderly Queue/Orderly Queue.cpp
class Solution {
public:
string orderlyQueue(string s, int k) {
 
if(k>1){
       sort(s.begin(),s.end());
       return s;
    }
    else{
        string res=s;
        
        for(int i=0;i<s.length();i++){
            s=s.substr(1)+s.substr(0,1);
            res=min(res,s);
        }
        return res;
    }
    
    return s;
    
}
};